Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -3312,9 +3320,26 @@ else if (thisReq_ instanceof ServerRequestLogout) {
                                     initState_ = SESSION_STATE.INITIALISED;
                                     // Publish success to listeners
                                     thisReq_.onRequestSucceeded(serverResponse, branchReferral_);
-
                                     isInitReportedThroughCallBack = ((ServerRequestInitSession) thisReq_).hasCallBack();
-                                    checkForAutoDeepLinkConfiguration();
+
+                                    boolean isActivityEnabledForPromoView = true;
+                                    boolean isPromoViewShowing = false;
+                                    PromoViewHandler promoViewHandler = PromoViewHandler.getInstance();
+                                    promoViewHandler.saveAppPromoViews();
+                                    if(currentActivityReference_.get() instanceof IPromoViewControl){
+                                        isActivityEnabledForPromoView = !((IPromoViewControl)currentActivityReference_.get()).skipPromoViewsOnThisActivity();
+                                    }
+                                    if(isActivityEnabledForPromoView) {
+                                        isPromoViewShowing = promoViewHandler.showPromoView(((ServerRequestInitSession) thisReq_).getPromoActonName(),
```

Comment:
is `showPromoView` ever going to be used anywhere else? I ask because you pass `Branch.this` for your interface callbacks - might be easier for `PromoViewHandler` class to keep reference as array of callbacks if this is to be used elsewhere